README.md

```diff
@@ -137,6 +137,99 @@ client.link_to(stream: 'anotherstream', events: [exisiting_event1, ...])
 When you read from stream where links are placed. By default Event Store Client always resolve links for you returning the event that points to the link. You can use the ES-ResolveLinkTos: false HTTP header during readin stream to tell Event Store Client to return you the actual link and to not resolve it.
 More info: [ES-ResolveLinkTos](https://eventstore.org/docs/http-api/optional-http-headers/resolve-linkto/index.html?tabs=tabid-1%2Ctabid-3).
 
+## Event Mappers
+
+We offer two types of mappers - default and encrypted.
+
+### Default Mapper
+
+This is used out of the box. It just translates the EventClass defined in your application to
+Event parsable by event_store and the other way around.
+
+### Encrypted Mapper
+
+This is implemented to match GDPR requirements. It allows you to encrypt any event using your
+encryption_key repository.
+
+```ruby
+mapper = EventStoreClient::Mapper::Encrypted.new(key_repository)
+EventStoreClient.configure do |config|
+  config.mapper = mapper
+end
+```
+
+The Encrypted mapper uses the encryption key repository to encrypt data in your events according to the event definition.
+
+Here is the minimal repository interface for this to work.
+
+```ruby
+class DummyRepository
+  class Key
+    attr_accessor :iv, :cipher, :id
+    def initialize(id:, **)
+      @id = id
+    end
+  end
+
+  def find(user_id)
+    Key.new(id: user_id)
+  end
+
+  def encrypt(*)
+    'darthvader'
+  end
+
+  def decrypt(*)
+    { first_name: 'Anakin', last_name: 'Skylwalker'}
+  end
+end
+```
+
+Now, having that, you only need to define the event encryption schema:
+
+```ruby
+class EncryptedEvent < EventStoreClient::DeserializedEvent
+  def schema
+    Dry::Schema.Params do
+      required(:user_id).value(:string)
+      required(:first_name).value(:string)
+      required(:last_name).value(:string)
+      required(:profession).value(:string)
+    end
+  end
+
+  def self.encryption_schema
+    {
+      key: ->(data) { data['user_id'] },
```

Comment:
why not just `key: 'user_id'` and look up the value later?

Reply:
I think that approach gives you more flexibility if you event needs some logic inside the lambda

Reply:
Not sure i understand. The event will not be changed later anyway...

Reply:
@ncri @aroeczek It's more like keeping the logic in the classes they belong to.

In this implementation when I want to extract the identifier, I rely on event implementation on how to store it.
I don't care if event needs any processing before giving me the key, nor which key it uses to retrieve it.

Just I moved the logic to deliver the encryption identifier the event where it belongs, so callers does not need to worry.